crates/transaction-generator-lib/src/p2p_transaction_generator.rs

```diff
@@ -91,6 +90,38 @@ impl P2PTransactionGenerator {
             },
         }
     }
+
+    /// Generate a given number (`num_txns`) of transactions that satisfies some constraints.
+    pub fn generate_block(&mut self, num_txns: usize, transfer_amount: u64, no_rw_conflict: bool, num_txns_per_sender: usize) -> Vec<(SignedTransaction, AccountAddress)> {
+        let num_accounts = self.accounts().len();
+        if no_rw_conflict {
+            // We need `num_txns` distinct senders and `num_txns` distinct recipients.
+            assert!(num_accounts >= num_txns * 2);
+            let indices = rand::seq::index::sample(&mut self.rng, self.accounts.len(), num_txns * 2);
```

Comment:
I might be wrong here but IIUC, this doesn't guarantee uniqueness, it is possible that the random selection will yield the same index twice.

Reply:
According to [the doc](https://docs.rs/rand/0.7.3/rand/seq/index/fn.sample.html) we don't need to worry. (I also tested a bit.)